### DISCUSSION TITLE
Retornar uma mensagem customizada no Authorize para "invalid token" ou "missed token"

### DISCUSSION THREAD

**Discussion by lucastatagiba** [balta-io/forum#485] [Tira Dúvidas] [answered]
Olá, boa tarde, nos meus controllers, tenho usado o atributo [Authorize], e quando o usuário insere um token invalido ou não insere o token no body da response não vem nada e o que vem é o status code 401 Unauthorized, que até então normal.
Porém eu gostaria de retornar no body da resposta, mensagens customizadas tipo: "Por favor insira um token"(quando ele não passa token) ou "token inválido"(quando ele passa um token inválido). Qual seria a melhor forma de fazer isso ?

**Comment by andrebaltieri** [comment, marked as answer]
Boa tarde, @lucastatagiba, como vai?

Você pode criar um middleware customizado para tratamento global de exceção e interceptar/mudar esta mensagem:
https://learn.microsoft.com/pt-br/aspnet/core/fundamentals/error-handling?view=aspnetcore-8.0#exception-handler-lambda
